### PR TITLE
LCM Add Tests

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR adds tests for LCM

### CODE REVIEW

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -1325,6 +1338,7 @@ def lora_state_dict(
                         weight_name = cls._best_guess_weight_name(
                             pretrained_model_name_or_path_or_dict, file_extension=".safetensors"
                         )
+                    print("weight_name", weight_name)
```
Comment (sayakpaul) [inline src/diffusers/loaders.py:1341]: Remove.

File: src/diffusers/loaders.py [hunk 2]
```diff
@@ -1411,6 +1451,15 @@ def _best_guess_weight_name(cls, pretrained_model_name_or_path_or_dict, file_ext
...
         )
 
+        if any(f.endswith(LORA_WEIGHT_NAME) for f in targeted_files):
+            targeted_files = list(
+                filter(lambda x: x.endswith(LORA_WEIGHT_NAME), targeted_files)
+            )
+        elif any(f.endswith(LORA_WEIGHT_NAME_SAFE) for f in targeted_files):
+            targeted_files = list(
+                filter(lambda x: x.endswith(LORA_WEIGHT_NAME_SAFE), targeted_files)
+            )
```
Comment (sayakpaul) [inline src/diffusers/loaders.py:1461]: Nice!

File: src/diffusers/utils/peft_utils.py [hunk 3]
```diff
@@ -204,6 +209,81 @@ def set_weights_and_activate_adapters(model, adapter_names, weights):
                 module.active_adapter = adapter_names
 
 
+def find_adapter_config_file(
```
Comment (sayakpaul) [inline src/diffusers/utils/peft_utils.py:212]: Very useful!

File: src/diffusers/loaders.py [hunk 4]
```diff
@@ -1622,7 +1671,11 @@ def load_lora_into_unet(
                 if "lora_B" in key:
                     rank[key] = val.shape[1]
 
-            lora_config_kwargs = get_peft_kwargs(rank, network_alphas, state_dict, is_unet=True)
+            if peft_config is not None:
+                lora_config_kwargs = LoraConfig.from_json_file(peft_config)
+            else:
+                lora_config_kwargs = get_peft_kwargs(rank, network_alphas, state_dict, is_unet=True)
+
```
Comment (sayakpaul) [inline src/diffusers/loaders.py:1678]: Shouldn't this come under an `if USE_PEFT_BACKEND`?

**Review by sayakpaul — COMMENTED** [review]
LGTM except https://github.com/huggingface/diffusers/pull/5707/files#r1387663161.

File: src/diffusers/loaders.py (hunk 4 above)
Comment (patrickvonplaten) [reply src/diffusers/loaders.py:1678]: Good point!

**Pull request merged**